src/quic/packet.cc

```diff
@@ -0,0 +1,406 @@
+#if HAVE_OPENSSL && NODE_OPENSSL_HAS_QUIC
+
+#include "packet.h"
+#include <base_object-inl.h>
+#include <crypto/crypto_util.h>
+#include <env-inl.h>
+#include <ngtcp2/ngtcp2.h>
+#include <ngtcp2/ngtcp2_crypto.h>
+#include <node_sockaddr-inl.h>
+#include <req_wrap-inl.h>
+#include <uv.h>
+#include <v8.h>
+#include <string>
+#include "bindingdata.h"
+#include "cid.h"
+#include "tokens.h"
+
+namespace node {
+
+using v8::FunctionTemplate;
+using v8::Local;
+using v8::Object;
+
+namespace quic {
+
+namespace {
+static constexpr size_t kRandlen = NGTCP2_MIN_STATELESS_RESET_RANDLEN * 5;
+static constexpr size_t kMinStatelessResetLen = 41;
+static constexpr size_t kMaxFreeList = 100;
+}  // namespace
+
+struct Packet::Data final : public MemoryRetainer {
+  MaybeStackBuffer<uint8_t, kDefaultMaxPacketLength> data_;
+
+  // The diagnostic_label_ is used only as a debugging tool when
+  // logging debug information about the packet. It identifies
+  // the purpose of the packet.
+  const std::string diagnostic_label_;
+
+  void MemoryInfo(MemoryTracker* tracker) const override {
+    tracker->TrackFieldWithSize("data", data_.length());
+  }
+  SET_MEMORY_INFO_NAME(Data)
+  SET_SELF_SIZE(Data)
+
+  Data(size_t length, const char* diagnostic_label)
+      : diagnostic_label_(diagnostic_label) {
```

Comment:
```suggestion
  Data(size_t length, const std::string& diagnostic_label)
      : diagnostic_label_(diagnostic_label) {
```
Since `diagnostic_label_`'s type is just `std::string`, though can't we use a `std::string_view` instead?